python/cugraph-pyg/cugraph_pyg/loader/cugraph_node_loader.py

```diff
@@ -253,55 +256,97 @@ def __next__(self):
             self.__data.dropna(inplace=True)
 
+            if (
+                len(self.__graph_store.edge_types) == 1
+                and len(self.__graph_store.node_types) == 1
+            ):
+                group_cols = ["batch_id", "hop_id"]
+                self.__data_index = self.__data.groupby(group_cols, as_index=True).agg(
+                    {"sources": "max", "destinations": "max"}
+                )
+                self.__data_index.rename(
+                    columns={"sources": "src_max", "destinations": "dst_max"},
+                    inplace=True,
+                )
+                self.__data_index = self.__data_index.to_dict(orient="index")
+
         # Pull the next set of sampling results out of the dataframe in memory
         f = self.__data["batch_id"] == self.__next_batch
         if self.__renumber_map is not None:
             i = self.__next_batch - self.__start_inclusive
-            ix = self.__renumber_map.iloc[[i, i + 1]]
-            ix_start, ix_end = ix.iloc[0], ix.iloc[1]
-            current_renumber_map = self.__renumber_map.iloc[ix_start:ix_end]
-            if len(current_renumber_map) != ix_end - ix_start:
-                raise ValueError("invalid renumber map")
-        else:
-            current_renumber_map = None
 
-        sampler_output = _sampler_output_from_sampling_results(
-            self.__data[f], current_renumber_map, self.__graph_store
-        )
+            # this should avoid d2h copy
+            current_renumber_map = self.__renumber_map[
+                self.__renumber_map_offsets[i] : self.__renumber_map_offsets[i + 1]
+            ]
 
-        # Get ready for next iteration
-        self.__next_batch += 1
+        else:
+            current_renumber_map = None
 
         # Get and return the sampled subgraph
-        if isinstance(torch_geometric, MissingModule):
-            noi_index, row_dict, col_dict, edge_dict = sampler_output["out"]
-            return _filter_cugraph_store(
-                self.__feature_store,
+        if (
+            len(self.__graph_store.edge_types) == 1
+            and len(self.__graph_store.node_types) == 1
+        ):
+            sampler_output = _sampler_output_from_sampling_results_homogeneous(
+                self.__data[f],
+                current_renumber_map,
                 self.__graph_store,
-                noi_index,
-                row_dict,
-                col_dict,
-                edge_dict,
+                self.__data_index,
+                self.__next_batch,
             )
         else:
-            out = torch_geometric.loader.utils.filter_custom_store(
-                self.__feature_store,
-                self.__graph_store,
-                sampler_output.node,
-                sampler_output.row,
-                sampler_output.col,
-                sampler_output.edge,
+            sampler_output = _sampler_output_from_sampling_results(
```

Comment:
I would strongly suggest renaming this function to `heterogeneous` to improve code readability.  
```suggestion
            sampler_output = _sampler_output_from_sampling_results_heterogeneous(
```

Reply:
done